### ISSUE TITLE
Double @@ from links.

### ISSUE THREAD

**Issue by p37307** [bramus/mastodon-profile-redirect#4]
When going from a profile toot/post, url adds a double @@ in the transformed url.

For example, go to a post, format like `https://TheirInstance/@user/109319548613954200` and click to have it transformed. The url becomes `https://Your-Instance/@@username@TheirInstance`. The proper URL should be `https://Your-Instance/@username@TheirInstance`.

Also, it would be nice to have the post URL transformed in the new link like `https://Your-Instance/@username@TheirInstance/109319548613954200`

Google Chrome Store Version 0.2.1, Updated November 23, 2022, running on Microsoft Edge Chrome Version 107.0.1418.56 (Official build) (64-bit) [Chromium]

Other than that, so far flawless. Great extension. [https://oldfriends.live/@paul](https://oldfriends.live/@paul)

**Comment by bramus** [comment]
Thanks for the report paul! I’ll take a look.

**Comment by bramus** [comment]
I tried to reproduce this but have not encountered the double `@`. I’ve noticed though that Mastodon not always injects the proper tags when navigating from a previous page (SPA) vs actually loading a page. I’ve tried both here, yet still haven’t encountered it. 

**I’ll close this for now, but if this happens again can you re-build the journey you took to reach the page, and post all those URLs here as a reply? Maybe then I can reproduce upon which I’ll gladly reopen.**

About redirecting from `https://TheirInstance/@user/109319548613954200` to `https://Your-Instance/@username@TheirInstance/109319548613954200`: this is not possible, as ids are different on instances.

See https://front-end.social/@bramus/109394868285321647 and https://mastodon.social/@bramus@front-end.social/109394868830041245. They’re both the same posts, with different IDs on each instance.

**Comment by chromaniac2** [comment]
It would be nice to get proper redirection support on post links as mentioned in above post. The bookmarklet I am using does that! 

https://homepages.inf.ed.ac.uk/imurray2/code/mastodon.html#:~:text=bookmarklet%20works%20everywhere.-,Instance%20changer,-%5BLast%20updated%202022

I came here to file a request for this but it's mentioned in this post so commenting instead. Thanks! 

----

The bookmark is able to handle post id difference somehow. For example. I used it on this post url: 

https://mastodon.social/@Hbomberguy/146524

And it redirected just fine to this post on my instance. 

https://social.vivaldi.net/@Hbomberguy@mastodon.social/96982677509470189

Maybe the code in it might help. Thanks!
